Src/library/net/abstract/socket/socket.e

```diff
@@ -514,6 +514,20 @@ feature -- Input
 			end
 		end
 
+	read_character_noexception
```

Comment:
At the Eiffel level, I'm not sure we want those in the EiffelNet classes yet. Maybe you can add them in a SOCKET descendant for EiffelWeb. Still put the new externals in the SOCKET class.